### PR TITLE
Fix: EncryptedField getDataForGrid return typehint is defined as array while it could returns string

### LINKED ISSUE #17777: [Bug]: EncryptedField getDataForGrid return typehint is defined as array while it could returns string
### Pimcore version

11

### Steps to reproduce

1. Create an EncryptedField with datatype "Input"
2. Add new object
3. Put some value in the EncryptedField
4. Display these objects in grid view
5. Add the EncryptedField in grid options

### Actual Behavior

Error is thrown : Message: Pimcore\Model\DataObject\ClassDefinition\Data\EncryptedField::getDataForGrid(): Return value must be of type array, string returned

### Expected Behavior

It should display the value as string

### PR DESCRIPTION
**Pull request by blankse**
## Changes in this pull request  
Resolves #17777

### CODE REVIEW

File: models/DataObject/ClassDefinition/Data/EncryptedField.php [hunk 1]
```diff
@@ -220,7 +206,7 @@ public function getDataFromEditmode(mixed $data, DataObject\Concrete $object = n
         return null;
     }
 
-    public function getDataFromGridEditor(float $data, Model\DataObject\Concrete $object = null, array $params = []): float|Model\DataObject\Data\EncryptedField
+    public function getDataFromGridEditor(mixed $data, Model\DataObject\Concrete $object = null, array $params = []): mixed
```
Comment (robertSt7) [inline models/DataObject/ClassDefinition/Data/EncryptedField.php:209]: @blankse I see in your PR some type changes of public non internal functions. Isn't that a BC-Break?

Comment (blankse) [reply models/DataObject/ClassDefinition/Data/EncryptedField.php:209]: @robertSt7 so what is your idea to fix this data type? Can't we use it until Pimcore 12?

Comment (blankse) [reply models/DataObject/ClassDefinition/Data/EncryptedField.php:209]: @robertSt7 Do you think we can merge this into 11.5 along with the Update Notes? I don't think anyone has extended this class with incorrect type hints. If they did, they should also encounter an error when trying to use the methods.

Comment (robertSt7) [reply models/DataObject/ClassDefinition/Data/EncryptedField.php:209]: @blankse You are right. I see what I can do

Comment (robertSt7) [reply models/DataObject/ClassDefinition/Data/EncryptedField.php:209]: @blankse I think we can merge this without Update Notes, since it fails until now

**Comment by robertSt7** [comment]
@blankse Thanks a lot for the fix and your contribution.

**Pull request merged**